example/src/app/utils/request.permissions.ts

```diff
@@ -0,0 +1,66 @@
+declare let cordova: any;
+
+export function requestPermissions(successCb: Function, errorCb: Function) {
+  var isAndroid =
+    navigator.userAgent.match(/Android/i) &&
+    navigator.userAgent.match(/Android/i)!.length > 0;
+  var isIOS = /iPhone|iPad|iPod/i.test(navigator.userAgent);
+
+  if (isAndroid) {
+    cordova.plugins.diagnostic.requestRuntimePermissions(
+      function (permissions: Map<string, string>) {
+        console.log('EXAMPLE> permissions statuses: ', permissions);
+        successCb();
+      },
+      function (error: any) {
+        errorCb(JSON.stringify(error));
+      },
+      [
+        // Android
+        // We ask this permission to obtain the GPS data of the user and paint it on our map.
+        cordova.plugins.diagnostic.permission.ACCESS_FINE_LOCATION,
+        // We ask this last 2 permissions to improve the accuracy of our SDK when locating the user.
+        cordova.plugins.diagnostic.permission.BLUETOOTH_CONNECT,
+        cordova.plugins.diagnostic.permission.BLUETOOTH_SCAN,
+      ]
+    );
+  } else if (isIOS) {
+    cordova.plugins.diagnostic.getLocationAuthorizationStatus(
+      (status: string) => {
+        if (status == 'authorized') {
+          successCb();
```

Comment:
Aquí chamas á callback de éxito, pero a execución continúa e fará o requestLocationAuthorization de máis abaixo. É o que queres?

Reply:
Probei a meter un log despois na linha 38 e parece ser que non chega a executar ese codigo, funcionaria como return nese caso.